### PR TITLE
.github/workflows: add codedov again

### PR DESCRIPTION
**Pull request by anonymouse64**
This now works, so we should be able to start getting coverage again.

I enabled coverage for all variants of the unit tests, it's unclear to me if we want coverage to be reported for the withbootassets and the nosecboot tags or not, I seem to recall things were slow if we enabled that, but maybe it's worth the slowness?

### CODE REVIEW

**Comment by anonymouse64** [comment]
Oops my PR here drops our code coverage by 2.4%! :smile:

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -209,8 +209,16 @@ jobs:
...
         ${{ github.workspace }}/bin/govendor remove github.com/snapcore/secboot
         ${{ github.workspace }}/bin/govendor remove +unused
-        # we don't need coverage anymore
-        SKIP_COVERAGE=1 SKIP_DIRTY_CHECK=1 GO_BUILD_TAGS=nosecboot ./run-checks --unit
+        SKIP_DIRTY_CHECK=1 GO_BUILD_TAGS=nosecboot ./run-checks --unit
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v2
+      with:
+        # token: ${{ secrets.CODECOV_TOKEN }} # needed ?
+        fail_ci_if_error: true
```
Comment (mvo5) [inline .github/workflows/test.yaml:217]: Why is this needed?

**Review by mvo5 — COMMENTED** [review]
Thanks, this looks super nice. Just one question inline and let's hope this `2%` drop message from coverage is a one time fluke. But worst case we just disable it again if it has too many false stats :)

**Review by mardy — APPROVED** [review]
Thanks a lot! :-)

Maybe the drop in coverage is because codecov remembers what was the last coverage level you had long ago before disabling it?

**Comment by mvo5** [comment]
I like it and I merged it now, very keen to see what it will report. Worst case is that we simply revert again if it's not useful :)

File: .github/workflows/test.yaml (hunk 1 above)
Comment (anonymouse64) [reply .github/workflows/test.yaml:217]: This makes the action fail the unit-tests action if there is an error uploading, otherwise it will just silently fail and not report back to us the coverage. I think this is good, since it means if we ever have problems uploading then we will find out about it via failing actions on PR's.

**Comment by anonymouse64** [comment]
@mvo5 indeed I think the drop in coverage is because the last commit codecov.io has coverage on is 8170654c737fc6fc93c03b8c44f5a08767d4fa38 (from the comment above)
